src/java.base/share/classes/java/io/Reader.java

```diff
@@ -207,6 +208,11 @@ public int read(char[] cbuf, int off, int len) throws IOException {
                 return n;
             }
 
+            public String readAllChars() throws IOException {
```

Comment:
The addition of this method effectively resolves [JDK-8309726](https://bugs.openjdk.org/browse/JDK-8309726), so I hope you don't mind me setting you as the assignee there, just to ensure we don't forget to close JDK-8309726 in turn once JDK-8354724 is closed? 🙂